javalin/src/main/java/io/javalin/json/JavalinGson.kt

```diff
@@ -6,15 +6,27 @@ import io.javalin.util.CoreDependency
 import io.javalin.util.DependencyUtil
 import io.javalin.util.JavalinLogger
 import io.javalin.util.Util
+import io.javalin.util.javalinLazy
 import java.io.BufferedWriter
 import java.io.InputStream
 import java.io.InputStreamReader
 import java.io.OutputStream
 import java.io.OutputStreamWriter
 import java.lang.reflect.Type
+import java.util.concurrent.ExecutorService
+import java.util.function.Supplier
 import java.util.stream.Stream
 
-open class JavalinGson(private val gson: Gson = Gson()) : JsonMapper {
+open class JavalinGson @JvmOverloads constructor(
+    private val gson: Gson = Gson(),
+    private var pipedStreamExecutorSupplier: (Supplier<ExecutorService>)? = null,
+) : JsonMapper {
+
+    private val pipedStreamExecutor by javalinLazy {
+        pipedStreamExecutorSupplier
+            ?.get()
+            ?: throw NotImplementedError("JavalinGson does not support piped streams. Use JavalinGson.create(Gson, pipedStreamExecutorSupplier)")
+    }
```

Comment:
Why do it like this?

Reply:
I'd need to add executor-service to each JavalinGson/JavalinJackson instance in tests (probably about 20+ occurrences), so I've made it optional for now. If you'd like to continue working on this PR, we could refactor it at the end.